TempSensor.ino

```diff
@@ -210,51 +292,73 @@ class MovingAverage {
       m_sum = m_sum + value;
       m_rotating_counter.increment();
     }
+    void reset(){
+      m_rotating_counter.resetCounter();
+    }
 };
 
 void setup() {
+  // set up trigger button
+  trigger.begin();
+  // set up laser
+  laser.begin();
+  // set up IR sensor
   mlx.begin();
   // set up the LCD's number of columns and rows:
   lcd.begin(16, 2); 
-  lcd.home ();
+  lcd.home();
 }
 
 
 MovingAverage avg_celsius;
 MovingAverage avg_fahrenheit;
 
 void loop() {
-  // Turn off the display:
-  lcd.noDisplay();
-  delay(refreshTempReading);
-
-  // Get temperature readings
-  int temp_celcius = mlx.readObjectTempC();
-  int temp_fahrenheit = mlx.readObjectTempF();
-
-  // Populate temperature values for moving average
-  if (flagMovingAverageEnable == true) {
-    avg_celsius.pushValue(temp_celcius);
-    avg_fahrenheit.pushValue(temp_fahrenheit);
-  }
+  int temp_celcius;
+  int temp_fahrenheit;
+
+  switch (trigger.readTriggerStatus()){
+    case LOW:
```

Comment:
Since we are not able to start the trigger with a setting of LOW in
https://github.com/CaringIndians/thermalsensor/pull/6/files#diff-a7e9cd7b80b8f4bf39578dbe1daea2b1R198

We need to treat LOW as the case when the temperature will be measured.

Is my understanding wrong here?

@yashtambi @kamatajinkya

Reply:
Yes

Reply:
Thanks for clarifying